tools/mason/MasonSearch.chpl

```diff
@@ -85,16 +91,21 @@ proc isHidden(name : string) : bool {
 proc findLatest(packageDir) {
   var ret = new VersionInfo(0, 0, 0);
   const suffix = ".toml";
-
+  var foundValidVersion = false;
   for fi in listdir(packageDir, files=true, dirs=false) {
-    assert(fi.endsWith(suffix));
-    const end = fi.length - suffix.length;
-    const ver = new VersionInfo(fi[1..end]);
-
-    if ver > ret then ret = ver;
+    if fi.endsWith(suffix) {
+      const end = fi.length - suffix.length;
+      const ver = new VersionInfo(fi[1..end]);
+      if ver > ret then ret = ver;
+      foundValidVersion = true;
+    }
+    else {
+      var warning_str = "file with bad/(not a .toml) extension encountered. skipping ";
+      warning_str += fi + "  in package " + packageDir;
+      warning(warning_str);
+    }
   }
-
-  return ret;
+  return (foundValidVersion, ret);
```

Comment:
What if there are no files inside the directory? Or there is only one file with a bad name like `0.1.0` instead of `0.1.0.toml` ?

Reply:
>  Or there is only one file with a bad name like 0.1.0 instead of 0.1.0.toml

This check above should prevent us from modifying `ret` when the file has a bad name:

```chpl
if fi.endsWith(suffix) {
  ...
}
```

> What if there are no files inside the directory?

Similarly, the function would return a `VersionInfo(0,0,0)` in this case, so we should still be OK.

Reply:
Does `VersionInfo(0, 0, 0)` mean the package will be skipped ?

Reply:
Yeah, I suggested we check if the the version is equal to `VersionInfo(0, 0, 0)` to determine if a valid package version was found, since `VersionInfo(0, 0, 0)` is not a legal version number.